src/utils/async.ts

```diff
@@ -48,3 +50,42 @@ export async function asyncIterableToArray<T>(
   const iter = source[Symbol.asyncIterator]();
   return asyncIteratorToArray(iter);
 }
+
+export function mapEitherAsyncIterator<E, T, V>(
+  iter: AsyncIterator<Either<E, T>, Either<E, T>>,
+  f: (t: T) => V
+): AsyncIterator<V> {
+  return {
+    next: async () => {
+      while (true) {
+        const { done, value } = await iter.next();
+        if (done) {
+          return { done, value: undefined };
+        }
+        if (isRight(value)) {
+          return { done, value: f(value.value) };
+        }
+      }
+    }
+  };
+}
+
+export async function flattenAsyncIterator<T>(
+  iter: AsyncIterator<ReadonlyArray<T>>
+): Promise<AsyncIterator<T>> {
+  const array = await asyncIteratorToArray(iter);
+  const flattenArray = array.reduce((acc, item) => {
+    return acc.concat(item);
+  }, []);
```

Comment:
doesn't this return the same as `array`? (it looks like an identity)

Reply:
With the `concat` method 2 ore more arrays are joined into flattenArray. It will be an identity if `push` was used instead of `concat`. Maybe is better `return [...acc, ...item]`?